srml/im-online/src/lib.rs

```diff
@@ -596,7 +596,7 @@ impl<T: Trait> support::unsigned::ValidateUnsigned for Module<T> {
 				priority: TransactionPriority::max_value(),
 				requires: vec![],
 				provides: vec![(current_session, authority_id).encode()],
-				longevity: TransactionLongevity::max_value(),
+				longevity: 128,
```

Comment:
Hmm, why not make this parameter configurable? Or make it proportional to the length of a session?

Whatever, but I would not hardcode this parameter.

Reply:
Fair enough. We can make a parameter for this I suppose, the actual session length is not easy to retrieve, as it's controlled by `ShouldEndSession` handler, so for instance for Kusama it's controlled by babe, but it's fine to just pass it as parameter.

Reply:
AFAIK the session length is fixed. You are right that we call into `Babe` to know if a session should end, but `Babe` internally does not change the length or whatever. 
Even if the session length is not 100% correct, it will not make anything worse in this use case here. The transaction would just be valid for a little bit more time than it should be.